src/models/rpc/mod.rs

```diff
@@ -26,3 +29,510 @@ pub enum NetworkRpcRequest {
     Stellar(StellarRpcRequest),
     Evm(EvmRpcRequest),
 }
+
+/// Converts a raw JSON-RPC request to the internal NetworkRpcRequest format.
+///
+/// This function takes a generic JSON-RPC request and transforms it into a strongly-typed
+/// internal representation based on the specified network type. Each network type (EVM, Solana, Stellar)
+/// has different serialization requirements and this function handles the conversion accordingly.
+///
+/// # Arguments
+///
+/// * `request` - A raw JSON-RPC request as a `serde_json::Value` containing the method,
+///   parameters, and other JSON-RPC fields
+/// * `network_type` - The target network type that determines how the request should be parsed
+///   and structured
+///
+/// # Returns
+///
+/// Returns a `Result` containing:
+/// - `Ok(JsonRpcRequest<NetworkRpcRequest>)` - Successfully converted request with the appropriate
+///   network-specific format
+/// - `Err(ApiError)` - Conversion failed due to missing required fields, invalid format, or
+///   serialization errors
+///
+/// # Examples
+///
+/// ```rust,ignore
+/// use serde_json::json;
+/// use crate::models::NetworkType;
+///
+/// let request = json!({
+///     "jsonrpc": "2.0",
+///     "method": "eth_getBalance",
+///     "params": ["0x742d35Cc6634C0532925a3b844Bc454e4438f44e", "latest"],
+///     "id": 1
+/// });
+///
+/// let result = convert_to_internal_rpc_request(request, &NetworkType::Evm)?;
+/// ```
+pub fn convert_to_internal_rpc_request(
+    request: serde_json::Value,
+    network_type: &crate::models::NetworkType,
+) -> Result<crate::domain::JsonRpcRequest<NetworkRpcRequest>, crate::models::ApiError> {
+    let jsonrpc = request
+        .get("jsonrpc")
+        .and_then(|v| v.as_str())
+        .unwrap_or("2.0")
+        .to_string();
+
+    let id = request.get("id").and_then(|v| v.as_u64()).unwrap_or(1);
```

Comment:
What if the `id` is not a number? I believe string is also allowed also https://www.jsonrpc.org/specification